.github/workflows/examples-on-label.yml

```diff
@@ -0,0 +1,27 @@
+---
+name: Test examples on label
+
+on:
+  pull_request:
+    types:
+      - labeled
+
+concurrency:
+  cancel-in-progress: true
+  group: >-
+    ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+
+jobs:
+  test-examples:
+    if: contains(github.event.pull_request.labels.*.name, 'run-examples')
```

Comment:
The workflow runs whenever _any_ label is added on a PR (cannot control this), but the
1. job is skipped if the labels do not contain the `run-examples` label
2. job runs if the labels contain the `run-examples` label

One can add multiple labels in any order and the job will run if `run-examples` is included.

Reply:
Considering how long the examples workflows take, this seems like a very reasonable approach!

Reply:
Does this run when a new commit is pushed to a PR already labelled as `run-examples`?

Reply:
That's a good point ^. We could possible have this action run on all PRs, but it only runs the action job in the presence of the `run-examples` label

Reply:
Now it does!